src/providers/mssql/qgsmssqlconnection.cpp

```diff
@@ -468,6 +468,74 @@ QList<QgsVectorDataProvider::NativeType> QgsMssqlConnection::nativeTypes()
          ;
 }
 
+QString QgsMssqlConnection::buildQueryForSchemas( const QString &connName )
+{
+  QgsSettings settings;
+
+  QString selectedSchemas;
+
+  QString databaseName = settings.value( QStringLiteral( "/MSSQL/connections/" ) + connName + "/database" ).toString();
+
+  bool schemaFilteringEnabled = settings.value( QStringLiteral( "/MSSQL/connections/" ) + connName + "/schemasFiltering" ).toBool();
+
+  if ( schemaFilteringEnabled )
+  {
+    QVariant schemaSettingsVariant = settings.value( QStringLiteral( "/MSSQL/connections/" ) + connName + "/schemasFiltered" );
+
+    if ( schemaSettingsVariant.type() == QVariant::Map )
+    {
+      QVariantMap schemaSettings = schemaSettingsVariant.toMap();
+      QVariantMap schemaSettingsForDatabase = schemaSettings.value( databaseName ).toMap();
+      //schema filter
+
+
+      QStringList schemaNames;
+      for ( const QString &schemaName : schemaSettingsForDatabase.keys() )
+      {
+        if ( schemaSettingsForDatabase.value( schemaName ).toBool() )
+        {
+          schemaNames.append( "'" + schemaName + "'" );
```

Comment:
We should use `QgsMssqlProvider::quotedValue( schemaName )` here instead of `"'" + schemaName + "'"` to make sure some characters are escaped properly (e.g. single quotes).